src/machine/board_rpi3.go

```diff
@@ -0,0 +1,182 @@
+// +build rpi3
+
+package machine
+
+import (
+	"device/arm"
+)
+
+//specific to raspberry pi 3
+var MemoryMappedIO = uintptr(0x3F000000)
+
+
+// XXX initialized in preinit so we avoid interp.Unsupported
+var Aux *AuxPeripheralsRegisterMap //= (*AuxPeripheralsRegisterMap)(unsafe.Pointer(MemoryMappedIO + 0x00215000))
+var GPIO *GPIORegisterMap //= (*GPIORegisterMap)(unsafe.Pointer(MemoryMappedIO + 0x00200000))
+var SysTimer *SysTimerRegisterMap //= (*SysTimerRegisterMap)(unsafe.Pointer(MemoryMappedIO + 0x3000))
+
+type AuxPeripheralsRegisterMap struct {
+	InterruptStatus           uint32 //0x00
```

Comment:
You can use volatile registers in order to ensure that semantics are preserved correctly. You can look at the device files in `device/sam` or similar for inspiration (after you have run `make gen-device`).

Reply:
Done. Much cleaner.